### PR TITLE
30 header component

### PR DESCRIPTION
**Pull request by Saeed99Madi**

### CODE REVIEW

**Review by mohmmed23 — DISMISSED** [review]

File: client/src/components/Header/Actions/FireCall/FireCall.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    margin: '0 0.5rem',
+    width: 25,
+    height: 25,
+    'margin-right': 'auto',
+  },
+  '@media screen and (max-width:450px)': {
+    width: 23,
+    height: 23,
+  },
+};
```
Comment (Amoodaa) [inline client/src/components/Header/Actions/FireCall/FireCall.tsx:22]: explore the `styled(Avatar)` syntax

File: client/src/services/ApiService.ts [hunk 2]
```diff
@@ -3,47 +3,47 @@ import axios, { AxiosRequestConfig, AxiosResponse, AxiosInstance } from 'axios';
...
-      this.axios = axios.create({
-        baseURL: process.env.REACT_APP_BASE_URL,
-        headers: {
-          Authorization: `Bearer ${JwtService.getToken()}`,
-          accept: 'application/json',
-          'Content-Type': 'application/json',
-        },
-      });
-    }
+  public static init():void {
```
Comment (Amoodaa) [inline client/src/services/ApiService.ts:8]: this screams code editor configuration problems, does this happen every PR?

File: src/config/HttpCode.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-unused-vars */
+export enum HttpStatusCode {
```
Comment (Amoodaa) [inline src/config/HttpCode.ts:2]: Why did you list everything, list only the thing you need

More over, Http status codes are something that every web developer should know, therefor no need for an enum :)

File: client/src/App.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+import { Header } from './components/Header/Header';
 
 const App = () => (
   <div>
     <PostsContext>
 
-      <Nav />
-
+      <Header />
       <Outlet />
```
Comment (Amoodaa) [inline client/src/App.tsx:12]: weird spacing

File: client/src/components/Header/Actions/Search/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// @flow
+// import React from 'react';
```
Comment (Amoodaa) [inline client/src/components/Header/Actions/Search/Search.tsx:2]: Why are you menitoning flow? you're using typescript

**Review by Amoodaa — CHANGES_REQUESTED** [review]
PR is very messy :)
it has so many changes that should have never been pushed in the first place, can you review your own PR and let me know if you think its a good PR?

File: client/src/components/Header/Actions/Actions.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const authButtons = true;
+  return (
+    <Box sx={ActionsStyles.actionsStyle}>
+      {search ? <Search /> : null}
+      {fireCall ? <FireCall /> : null}
+      {userTicket ? <UserTicket /> : null}
+      {authButtons ? <AuthButtons /> : null}
+    </Box>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Saeed99Madi) [reply client/src/components/Header/Actions/Actions.tsx:23]: in progress code,
i shouldn't create pull request until i finish actually 
@Amoodaa

[the remark this answers is not in the document]
Comment (Amoodaa) [reply client/src/components/Header/Actions/Actions.tsx:23]: Convert them to draft PRs

**Review by mohmmed23 — APPROVED** [review]

File: client/src/services/debounce.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const depounce = (fn:Function, delay:number):Function => {
```
Comment (Amoodaa) [inline client/src/services/debounce.ts:1]: typo

File: client/package.json [hunk 8]
```diff
@@ -13,6 +13,7 @@
     "@fortawesome/react-fontawesome": "^0.2.0",
     "@mui/icons-material": "^5.10.9",
     "@mui/material": "^5.10.9",
+    "@mui/styled-engine-sc": "^5.10.6",
```
Comment (Amoodaa) [inline client/package.json:16]: why is this needed

File: client/src/components/Header/Actions/AuthButtons/AuthButtons.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  },
+};
+const signIn = {
+  color: '#D53449',
+  'border-color': '#D53449',
+  '@media screen and (max-width:450px)': {
+    'font-size': '0.45rem',
+    padding: '0.1rem',
+  },
+};
```
Comment (Amoodaa) [inline client/src/components/Header/Actions/AuthButtons/AuthButtons.tsx:32]:
```ts
const LoginButton = styled(Button)(`
... css ...
`);

<LoginButton onClick={...}>
```

File: client/src/components/Header/Actions/FireCall/style.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    margin: '0 0.5rem',
+    width: 25,
+    height: 25,
+    'margin-right': 'auto',
+  },
+  '@media screen and (max-width:450px)': {
+    width: 23,
+    height: 23,
+  },
+};
```
Comment (Amoodaa) [inline client/src/components/Header/Actions/FireCall/style.ts:17]: same as above

File: client/src/components/Header/Actions/Search/Search.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+  // change Context value
+  useEffect(() => {
+    depounce(() => {
+      if (setFilterObj) {
+        setFilterObj({ ...filterObj, content: searchInput });
+      }
+    }, 1500);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchInput]);
```
Comment (Amoodaa) [inline client/src/components/Header/Actions/Search/Search.tsx:35]: VERY VERY VERY BAD PATTERN
you can simply do it in the event handler
```ts
  const handleSearchInputChange = (event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
    setSearchInput(event.target.value);
    setFilterObj({ ...filterObj, content: searchInput });
  };
  ```

File: client/src/components/Header/NavHeader/styles.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  '@media screen and (max-width: 850px)': {
+    display: 'block',
+    color: '#222',
+    position: 'absolute',
+    top: '0',
+    right: '0',
+    'font-size': '1.3em',
+    margin: '10px',
+    cursor: 'pointer',
+  },
```
Comment (Amoodaa) [inline client/src/components/Header/NavHeader/styles.ts:31]: same as above

**Review by Amoodaa — CHANGES_REQUESTED** [review]

File: client/src/services/debounce.ts (hunk 7 above)
Comment (Saeed99Madi) [reply client/src/services/debounce.ts:1]: what do you mean of type ?

File: client/package.json (hunk 8 above)
Comment (Saeed99Madi) [reply client/package.json:16]: It's designed to use styled-components as the main styled engine instead of @emotion/styled.

File: client/src/components/Header/Actions/AuthButtons/AuthButtons.tsx (hunk 9 above)
Comment (Saeed99Madi) [reply client/src/components/Header/Actions/AuthButtons/AuthButtons.tsx:32]: great

File: client/src/components/Header/Actions/Search/Search.tsx (hunk 11 above)
Comment (Saeed99Madi) [reply client/src/components/Header/Actions/Search/Search.tsx:35]: i am trying to change in `filterObj` when user remove his hand form the keyboard,

File: client/src/components/Header/NavHeader/styles.ts (hunk 12 above)
Comment (Saeed99Madi) [reply client/src/components/Header/NavHeader/styles.ts:31]: great

**Review by mohmmed23 — APPROVED** [review]
